core/src/main/java/io/parsingdata/metal/data/Source.java

```diff
@@ -16,16 +16,11 @@
 
 package io.parsingdata.metal.data;
 
-import java.io.IOException;
 import java.math.BigInteger;
 
 public abstract class Source {
```

Comment:
I'm still a bit conflicted about this `abstract class`. The only reason that it is `abstract`, and not an `interface`, is that we can make the methods `protected`, correct?

I'd still prefer to use `interface` for something that is clearly an `interface`.

Reply:
I have an intuitive preference for interface as well. I think that if we end up not using the visibility effects for anything of real practical value, it will be better to convert it to an interface.